### DISCUSSION TITLE
Cannot add tolerations/node selectors/resources to CR

### DISCUSSION THREAD

**Discussion by emmahsax** [kiali/kiali#7206] [Q&A]
### Describe the bug

I am having trouble adding tolerations, node selectors, and resources to the CR, based on docs from:
* https://v1-78.kiali.io/docs/configuration/kialis.kiali.io/
* https://github.com/kiali/kiali-operator/blob/master/crd-docs/cr/kiali.io_v1alpha1_kiali.yaml

I am trying to apply the CR with Terraform like this:

```tf
resource "kubectl_manifest" "kiali" {
  yaml_body = <<-EOT
  apiVersion: kiali.io/v1alpha1
  kind: Kiali
  metadata:
    name: kiali
    namespace: istio-system
  spec:
    deployment:
      node_selector:
        mywebsite.com/nodegroup: worker
      resources:
        limits:
          cpu: 256m
          memory: 256Mi
        requests:
          cpu: 100m
          memory: 128Mi
      tolerations:
      - effect: NoSchedule
        key: mywebsite.com/nodegroup
        operator: Equal
        value: worker
    external_services:
      grafana:
        enabled: false
      istio:
        component_status:
          components:
          - app_label: istiod
            is_core: true
            is_proxy: false
          - app_label: ingressgateway
            is_core: true
            is_proxy: true
            namespace: istio-system
          - app_label: egressgateway
            is_core: false
            is_proxy: true
            namespace: istio-system
      prometheus:
        thanos_proxy:
          enabled: true
        url: http://prometheus-operated.prometheus:9090
      tracing:
        enabled: true
        in_cluster_url: http://tempo-query-frontend.opentelemetry:3100
        provider: tempo
        use_grpc: false
  EOT
}
```

I can't figure out why the kiali-operator isn't picking up those deployment settings. Here's the operator:

```tf
resource "helm_release" "kiali_operator" {
  chart      = "kiali-operator"
  name       = "kiali"
  namespace  = "istio-system"
  repository = "https://kiali.org/helm-charts"
  version    = var.kiali_helm_chart_version

  values = [
    <<-YAML
    nodeSelector:
      ${local.node_selectors_yaml}
    resources:
      ${indent(2, local.resources_yaml)}
    tolerations:
      ${indent(2, local.tolerations_yaml)}
    YAML
  ]
}

```

### What are the steps to reproduce this bug?

1. Create a kiali operator using the terraform block above
2. Create the CR using the terraform block above
3. Does it properly accept the toleration, node selector, or resources?

### Environment
*Learn about how to determine versions [here](https://kiali.io/docs/faq/general/#how-do-i-determine-what-version-i-am-running).*

- **Kiali version:** 1.78.0
- **Istio version:** 1.17.1
- **Kubernetes impl:** 
- **Kubernetes version:** 1.27
- **Other notable environmental factors:**

**Comment by jmazzitelli** [comment]
(_converted to Q&A discussion; this is likely not a bug_)

There are several things.

> * **Kiali version:** 1.78.0
> * **Istio version:** 1.17.1

I don't know if this is your issue, but I can tell you that your Kiali version is incompatible with the Istio version you are using. That combination is unsupported and I would be surprised if it works.

You should be at no higher than 1.66.1.

See: https://kiali.io/docs/installation/installation-guide/prerequisites/

```
Istio	Kiali Min	Kiali Max	Notes
1.20	1.76.0		
1.19	1.72.0	1.75.0	
1.18	1.67.0	1.73.0	
1.17	1.63.2	1.66.1	Avoid 1.63.0,1.63.1 due to a regression.
```

That said, you can use our verification tool to check the validity of your Kiali CR - things like this usually mean the Kiali CR is not formatted correctly. See https://kiali.io/docs/configuration/kialis.kiali.io/#validating-your-kiali-cr

Because you are using Terraform, we can't guarantee that correctness of what you are doing. Try to install using the supported, documented commands in the Installation docs. If it works, then something is wrong  with the Terraform setup.

Lastly, I can't tell, but it looks like you are configuring those resources for the operator and you are thinking it is translating those to the server. That is not how it works. You can set different resources for the operator (so when you set the resources in the operator helm chart, it is the operator pod that gets those settings). Then in the Kiali CR - that's the thing the operator processes and uses its config to set up the server pod (so its the Kiali CR's resources that go to the server pod). Again, I'm confused with our Terraform scripts (I'm not familiar with Terraform) so I could be wrong. But just realize that setting those values in the `resource "helm_release" "kiali_operator"` (the operator helm chart) sets the resources for the operator itself only (NOT the server).

**Comment by emmahsax** [comment]
I downgraded the version of Kiali to 1.66 as suggested. Then I deleted the kiali pod, hoping that a new one would pop up, and it never did. 🤦🏼 I'll have to ask my team what docs they used to install it initially.

**Comment by emmahsax** [comment]
So, I managed to get a kiali pod to come back by essentially removing those three settings, and then letting it come back. Then I brought the settings back one at a time, and eventually they took. Strange. But okay. The issue was probably the version.
